src/liballoc/tests/btree/map.rs

```diff
@@ -788,7 +788,10 @@ fn test_clone() {
 #[test]
 fn test_clone_from() {
     let mut map1 = BTreeMap::new();
+    #[cfg(not(miri))] // Miri is too slow
     let size = 30;
+    #[cfg(miri)]
+    let size = 13;
```

Comment:
I'm not really sure that testing 900 different variations of clone_from is buying us anything anyway, so maybe just go with 13 always?

Reply:
I agree. 0, 1, 10, 11 and 12 are the only interesting sizes for size=30 (note that `size` itself isn't tested). Next stop would be 144 (3 levels, not all edges pointing to leaves).

Also, as an aside, by always adding greater elements on both sides, I think (am not quite sure) that some kinds of trees are ignored (as in right leaf bigger than left leaf or so). I am pretty sure that you can have an 11 element tree packed in a lonely root or spread over 2 levels, depending on whether you remove the 6th or 7th element from a 12 element tree. But it's hard to diagnose since tests can't tell the difference. So other test cases don't try to go there either.

Reply:
The same can be said about `test_clone` (possibly an inspiration for this or future tests): if we can clone a 12 element (2 level, 2 leaf) tree, surely we can handle every bigger 2 level tree... but can we handle bigger? Yes, we can, and Miri can handle it fine too. Just don't ask it to clone every size of tree in between.